kube-core/src/schema.rs

```diff
@@ -36,6 +37,110 @@ impl Visitor for StructuralSchemaRewriter {
+                //
+                // This produces the following invalid CRD (snippet)
+                // "complexEnum": {
+                //     "type": "object",
+                //     "properties": {
+                //         "variantOne": {
+                //             "type": "object",
+                //             "properties": {
+                //                 "int": {
+                //                     "type": "integer",
+                //                     "format": "int32"
+                //                 }
+                //             },
+                //             "required": ["int"],
+                //         },
+                //         "VariantTwo": {
+                //             "type": "object",
+                //         }
+                //     },
+                //     "oneOf": [
+                //     {
+                //         "required": ["variantOne"],
+                //         "description": "First variant with an int"
+                //     },
+                //     {
+                //         "required": ["variantTwo"],
+                //         "description": "Second variant which doesn't has an attribute"
+                //     }
+                //     ],
+                //     "description": "This is a complex enum"
+                // }
+                //
+                // The correct solution requires to move the description from the oneOf section to to the complexEnum.properties
+                // The corrected solution looks like follows
+                // "complexEnum": {
+                //     "type": "object",
+                //     "properties": {
+                //         "variantOne": {
+                //             "type": "object",
+                //             "properties": {
+                //                 "int": {
+                //                     "type": "integer",
+                //                     "format": "int32"
+                //                 }
+                //             },
+                //             "required": ["int"],
+                //             "description": "First variant with an int"
+                //         },
+                //         "variantTwo": {
+                //             "type": "object",
+                //             "description": "Second variant which doesn't has an attribute"
+                //         }
+                //     },
+                //     "oneOf": [
+                //     {
+                //         "required": ["variantOne"],
+                //     },
+                //     {
+                //         "required": ["variantTwo"],
+                //     }
+                //     ],
+                //     "description": "This is a complex enum"
+                // }
+                //
+                // The following lines move the descriptions to the correct places
+                if let Schema::Object(SchemaObject {
```

Comment:
This could probably be integrated into the `if let` below.

Reply:
You mean integrating into the following `if let`? I tried this at first but by requiring the metadata to be `Some` it changes existing behavior and brakes tests.
```
                if let Schema::Object(SchemaObject {
                    instance_type: variant_type,
                    object: Some(variant_obj),
                    ..
                }) = variant
```

Reply:
Yeah, but we can export `metadata` from that condition and then just have a simpler `if let Some(variant_meta) = variant_meta {}` block inside. Anyway, not a huge problem either way.

Reply:
Thanks, missed that